qiskit/algorithms/evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -0,0 +1,124 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for solving linear equations for Quantum Time Evolution."""
+
+from typing import Union, List, Dict, Optional, Callable
+
+import numpy as np
+from qiskit.circuit import Parameter
+from qiskit.opflow import CircuitSampler, OperatorBase
+from ..calculators.metric_tensor_calculator import (
+    eval_metric_result,
+)
+from ..calculators.evolution_grad_calculator import (
+    eval_grad_result,
+)
+
+
+class VarQTELinearSolver:
+    """Class for solving linear equations for Quantum Time Evolution."""
+
+    def __init__(
+        self,
+        metric_tensor: OperatorBase,
+        evolution_grad: OperatorBase,
+        lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] = np.linalg.lstsq,
+        circuit_sampler: Optional[CircuitSampler] = None,
+        imag_part_tol: float = 1e-7,
+    ) -> None:
+        """
+        Args:
+            metric_tensor: A parametrized operator that represents the left-hand side of an ODE.
+            evolution_grad: A parametrized operator that represents the right-hand side of an ODE.
+            lse_solver: Linear system of equations solver that follows a NumPy
+                ``np.linalg.lstsq`` interface.
+            circuit_sampler: Samples circuits using an underlying backend.
+            imag_part_tol: Allowed value of an imaginary part that can be neglected if no
+                imaginary part is expected.
+        """
+
+        self._metric_tensor = metric_tensor
+        self._evolution_grad = evolution_grad
+        self._lse_solver = lse_solver
+        self._circuit_sampler = circuit_sampler
+        self._imag_part_tol = imag_part_tol
+
+    def solve_sle(
```

Comment:
Minor point, but could we rename this to `solve_lse` to be consistent with `lse_solver` and the docs which use `LSE` as abbreviation? 🙂